File: src/main/java/com/iponchallenge/entity/Challenge.java
package com.iponchallenge.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "challenges")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Challenge {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false)
    private String description;

    @Column(name = "target_days", nullable = false)
    private int targetDays;

    @Column(name = "reward_xp", nullable = false)
    private int rewardXp;

    // NO_SPEND | SAVINGS_TARGET | STREAK
    @Column(nullable = false)
    private String type;

    // Only used for SAVINGS_TARGET type
    @Column(name = "target_amount", precision = 12, scale = 2)
    private BigDecimal targetAmount;

    @Column(nullable = false)
    @Builder.Default
    private boolean active = true;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;
}

File: frontend/src/types/challenge.ts
export type ChallengeType = 'NO_SPEND' | 'SAVINGS_TARGET' | 'STREAK';

export interface Challenge {
  id: string;
  title: string;
  description: string;
  targetDays: number;
  rewardXp: number;
  type: ChallengeType;
  targetAmount: number | null;
  active: boolean;
  joined: boolean;
  completed: boolean;
  currentProgress: number;
  progressPercentage: number;
  startDate: string | null;
  completedAt: string | null;
}
